subprojects/language-native/src/main/java/org/gradle/language/cpp/plugins/CppExecutablePlugin.java

```diff
@@ -76,6 +78,8 @@ public void apply(final ProjectInternal project) {
         project.getComponents().add(application);
         project.getComponents().add(application.getDebugExecutable());
         project.getComponents().add(application.getReleaseExecutable());
+        project.getComponents().add(application.getDebugStaticExecutable());
```

Comment:
We don't want to add 'link everything statically' variants as a convention for applications. This (usually) doesn't make sense. That doesn't mean that this never happens, but for that situation we only need to make it possible to configure the build so that it will produce a 'link everything statically' variant, not the default.